src/mame/includes/nes.h

```diff
@@ -47,28 +47,42 @@
 #define NES_BATTERY 0
 #define NES_WRAM 1
 
+// so that the NES and Famiclones (VT03 for example) can use some common functionality
+class nes_base_state : public driver_device
+{
+public:
+	nes_base_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_ctrl1(*this, "ctrl1"),
+		m_ctrl2(*this, "ctrl2")
+	{ }
+
+	required_device<cpu_device> m_maincpu;
+	required_device<nes_control_port_device> m_ctrl1;
+	required_device<nes_control_port_device> m_ctrl2;
 
-class nes_state : public driver_device
+	DECLARE_READ8_MEMBER(nes_in0_r);
+	DECLARE_READ8_MEMBER(nes_in1_r);
+	DECLARE_WRITE8_MEMBER(nes_in0_w);
+};
+
+class nes_state : public nes_base_state
 {
 public:
 	nes_state(const machine_config &mconfig, device_type type, const char *tag)
-		: driver_device(mconfig, type, tag)
-		, m_maincpu(*this, "maincpu")
-		, m_ppu(*this, "ppu")
-		, m_ctrl1(*this, "ctrl1")
-		, m_ctrl2(*this, "ctrl2")
-		, m_exp(*this, "exp")
-		, m_cartslot(*this, "nes_slot")
-		, m_disk(*this, "disk")
-	{
-	}
+		: nes_base_state(mconfig, type, tag),
```

Comment:
Could you please at least make indentation consistent within a file?  the indentation of this constructor's initialisation list is different to the one for the class immediately above it.